k9mail/src/main/res/values/strings.xml

```diff
@@ -1192,33 +1192,42 @@ Please submit bug reports, contribute new features and ask questions at
-    <string name="crypto_msg_sign_mismatch">Signature made by wrong key.</string>
-    <string name="crypto_msg_sign_expired">Signature made by expired key.</string>
-    <string name="crypto_msg_sign_revoked">Signature made by revoked key.</string>
-    <string name="crypto_msg_sign_insecure">Signature made by insecure key.</string>
-    <string name="crypto_msg_sign_incomplete">Message must be fully downloaded to verify its signature.</string>
-
-    <string name="crypto_msg_signed_encrypted">Message is signed and encrypted.</string>
-    <string name="crypto_msg_unsigned_encrypted">There is no signature, this message may have been intercepted.</string>
-
+    <string name="crypto_msg_title_plaintext">Plaintext</string>
+    <string name="crypto_msg_unencrypted_sign_error">End-to-End signature contained an error</string>
+    <string name="crypto_msg_incomplete_signed">must fully download to process End-to-End signature</string>
+    <string name="crypto_msg_unsupported_signed">contains unsupported End-to-End signature</string>
+
+    <string name="crypto_msg_title_unencrypted_signed_e2e">End-to-End signed plaintext</string>
+    <string name="crypto_msg_unencrypted_sign_verified">from verified End-to-End key</string>
+
+    <string name="crypto_msg_title_unencrypted_signed">Signed plaintext</string>
+    <string name="crypto_msg_unencrypted_sign_mismatch">but End-to-End key didn\'t match sender</string>
+    <string name="crypto_msg_unencrypted_sign_expired">but End-to-End key is expired</string>
+    <string name="crypto_msg_unencrypted_sign_revoked">but End-to-End key is revoked</string>
+    <string name="crypto_msg_unencrypted_sign_insecure">but End-to-End key is not considered secure</string>
+    <string name="crypto_msg_unencrypted_sign_unknown">from unknown End-to-End key</string>
+
+    <string name="crypto_msg_title_encrypted_unknown">Encrypted</string>
+    <string name="crypto_msg_encrypted_error">but there was a decryption error</string>
+    <string name="crypto_msg_encrypted_incomplete">must be fully downloaded for decryption</string>
+    <string name="crypto_msg_encrypted_no_provider">but no crypto app is configured</string>
+    <string name="crypto_msg_encrypted_unsupported">but in an unsupported format</string>
+    <string name="crypto_msg_encrypted_cancelled">but decryption was cancelled</string>
+
+    <string name="crypto_msg_title_encrypted_unsigned">Encrypted</string>
+    <string name="crypto_msg_encrypted_unsigned">but not End-to-End</string>
+
+    <string name="crypto_msg_title_encrypted_signed_e2e">End-to-End encrypted</string>
+    <string name="crypto_msg_encrypted_sign_verified">from verified End-to-End key</string>
+
+    <string name="crypto_msg_title_encrypted_signed">Encrypted</string>
+    <string name="crypto_msg_encrypted_sign_unknown">from unknown End-to-End key</string>
+    <string name="crypto_msg_encrypted_sign_mismatch">but End-to-End key didn\'t match sender</string>
```

Comment:
'from someone else' might be clearer.

Reply:
not happy with this one either. "from someone else" sounds less "faulty" to me, it kind of implies that it's just "valid for a different person", not that it's "invalid for this message"?

Reply:
Yeah I guess you need a negative in there. Fine for now I think. We can always bikeshed and merge a tweak.